megatron/core/dist_checkpointing/strategies/torch.py

```diff
@@ -382,27 +380,27 @@ def save(self, sharded_state_dict: ShardedStateDict, checkpoint_dir: Path):
         # Use PyT saving mechanism
         save_state_dict(
             pyt_state_dict,
-            FileSystemWriter(checkpoint_dir, thread_count=self.thread_count),
+            FsspecWriter(path=checkpoint_dir, thread_count=self.thread_count),
```

Comment:
This need to be configurable, because I think Fsspec is slower than FilesystemWriter. Also, please note that there is a change on `main` which uses a custom `save_state_dict` and a custom FileSystemAsyncWriter for performance.